icu4c/source/common/cmemory.h

```diff
@@ -323,6 +323,11 @@ class MaybeStackArray {
      * @return the array pointer
      */
     T *getAlias() const { return ptr; }
+    /**
+     * Access without ownership change.
+     * @return the array pointer
+     */
+    const T *getConstAlias() const { return ptr; }
```

Comment:
As the `getAlias()` method already is marked `const`, these two statements would have the same result:

```
const T* p1 = msa.getAlias();
const T* p1 = msa.getConstAlias();
```

As would these:

```
const auto* p1 = msa.getAlias();
const auto* p1 = msa.getConstAlias();
```

Do you have a use-case where this new `getConstAlias()` method makes it possible to do something that itsn't already possible to do with the existing `getAlias()` method?

It is a bit unusual that the existing `getAlias()` method is `const` but returns a pointer to a non-`const` object. The PR description and the associated ticket are both very vague on what the purpose of these new methods are, but they both mention _const correctness_.

Might it be that what you really would want to do here is to change the existing `getAlias()` method to make it non-`const` and then add an overload that is `const` and returns a pointer to a `const` object?

That'd be a rather common pattern used to ensure _const correctness_ by requiring that the contents of the container be modified only through a non-`const` reference to the container object:

```
T *getAlias() { return ptr; }
const T *getAlias() const { return ptr; }
```